metrics/density/fast_footprint.sh

```diff
@@ -0,0 +1,556 @@
+#!/bin/bash
+# Copyright (c) 2017-2018 Intel Corporation
+# 
+# SPDX-License-Identifier: Apache-2.0
+#
+# A script to gather memory 'footprint' information as we launch more
+# and more containers
+# It allows configuration of a number of things:
+# - which container workload we run
+# - which container runtime we run with
+# - when do we terminate the test (cutoff points)
+#
+# There are a number of things we may wish to add to this script later:
+# - sanity check that the correct number of runtime components (qemu, shim etc.)
+#  are running at all times
+# - some post-processing scripts to generate stats and graphs
+#
+# The script gathers information about both user and kernel space consumption
+# Output is into a .json file, named using some of the config component names
+# (such as footprint-busybox.json)
+
+# Pull in some common, useful, items
+SCRIPT_PATH=$(dirname "$(readlink -f "$0")")
+source "${SCRIPT_PATH}/../lib/common.bash"
+
+# Note that all vars that can be set from outside the script (that is,
+# passed in the ENV), use the ':-' setting to allow being over-ridden
+
+# Default sleep for 3s to let containers come up and finish their
+# initialisation before we take the measures. Some of the larger
+# containers can take a number of seconds to get running.
+PAYLOAD_SLEEP="${PAYLOAD_SLEEP:-10}"
+
+# How long, in seconds, do we wait for KSM to 'settle down', before we
+# timeout and just continue anyway.
+KSM_WAIT_TIME="${KSM_WAIT_TIME:-300}"
+
+# How long, in seconds, do we poll for docker to complete launching all the
+# containers?
+DOCKER_POLL_TIMEOUT="${DOCKER_POLL_TIMEOUT:-300}"
+
+# How many containers do we launch in parallel before taking the PAYLOAD_SLEEP
+# nap
+PARALLELISM="${PARALLELISM:-10}"
+
+### The default config - run a small busybox image
+# Define what we will be running (app under test)
+#  Default is we run busybox, as a 'small' workload
+PAYLOAD="${PAYLOAD:-busybox}"
+PAYLOAD_ARGS="${PAYLOAD_ARGS:-tail -f /dev/null}"
+PAYLOAD_RUNTIME_ARGS="${PAYLOAD_RUNTIME_ARGS:- -m 2G}"
```

Comment:
I think we've discussed this before so although not required for this PR, it would be good if we could:

- Document the "minimum spec" for a metrics test system.
- Add checks to ensure this default value is possible on the host the test is running on.